samples/common.hpp

```diff
@@ -104,6 +104,23 @@ struct Log2<1>
     static constexpr uint value = 0;
 };
 
+// Matrix data initialization
+template <typename DataT>
+__host__ static inline void fill(DataT* mat, uint32_t m, uint32_t n)
+{
+    auto ld = n;
+    for(int i = 0; i < m; ++i)
+    {
+        for(int j = 0; j < n; ++j)
+        {
+            // Ascending order for each neighboring element.
+            // Alternate sign for even / odd
+            auto value      = (i * ld + j) % 17;
+            mat[i * ld + j] = (value % 2) ? -static_cast<DataT>(value) : static_cast<DataT>(value);
```

Comment:
May want to do a check std::is_signed<DataT> before attempting the sign inversion